engines/cryomni3d/versailles/documentation.cpp

```diff
@@ -0,0 +1,2070 @@
+				if (boxes.hitTest(0, mouse)) {
+					// Back in history
+					action = 0;
+				} else if (boxes.hitTest(1, mouse)) {
+					// Handle summary menu
+					Common::StringArray items;
+					items.push_back((*_messages)[61]);
+					items.push_back((*_messages)[62]);
+					unsigned int selectedItem = handlePopupMenu(surface, boxes.getBoxOrigin(1), false, 20, items);
+					if (selectedItem == 0) {
+						action = 1;
+					} else if (selectedItem == 1) {
+						action = 7;
+					}
+				} else if (boxes.hitTest(4, mouse)) {
+					// Next
+					action = 4;
+				} else if (boxes.hitTest(5, mouse)) {
+					// Previous
+					action = 5;
+				} else if (boxes.hitTest(6, mouse)) {
+					// Handle quit menu
+					Common::StringArray items;
+					items.push_back((*_messages)[60]);
+					unsigned int selectedItem = handlePopupMenu(surface, boxes.getBoxOrigin(6), false, 20, items);
+					if (selectedItem == 0) {
+						action = 6;
+					}
+				} else if (_currentHasMap && boxes.hitTest(8, mouse)) {
+					// Map
+					action = 8;
+				} else if (boxes.hitTest(9, mouse)) {
+					// Category name
+					action = 9;
+				} else if (_currentInTimeline && hoveredBox != -1u) {
+					// Clicked on a timeline entry
+					nextRecord = "VT";
+					nextRecord += kTimelineEntries[hoveredBox - 10].year;
+					// Fake a global jump
+					action = 3;
+				}
+			}
+			if (action == 4 || action == 5) {
+				if (action == 4 && _currentRecord == _categoryEndRecord) {
+					action = -1;
+					continue;
+				}
+				if (action == 5 && _currentRecord == _categoryStartRecord) {
+					action = -1;
+					continue;
+				}
+				Common::HashMap<Common::String, RecordInfo>::iterator hmIt = _records.find(_currentRecord);
+				if (hmIt == _records.end()) {
+					// Shouldn't happen
+					action = -1;
+					continue;
+				}
+				unsigned int recordId = hmIt->_value.id;
+				if (action == 4) {
+					recordId++;
+				} else if (action == 5) {
+					recordId--;
+				}
+				assert(recordId < _recordsOrdered.size());
+				nextRecord = _recordsOrdered[recordId];
+				// Fake a global jump
+				action = 3;
+			}
+		}
+	}
+
+	g_system->showMouse(false);
+	_engine->setCursor(181);
+	return action;
+}
+
+Common::String Versailles_Documentation::docAreaHandleGeneralMap() {
+	struct Area {
+		Common::Rect areaPos;
+		const char *record;
+		const char *bmp;
+		unsigned int messageId;
+		const Common::String *message;
+		Common::Point messagePos;
+		Graphics::Surface highlightedImg;
+
+		Area(const Common::Point &areaPos_, const char *bmp_, unsigned int messageId_,
+		     const char *record_ = nullptr) :
+			areaPos(areaPos_.x, areaPos_.y, areaPos_.x, areaPos_.y), record(record_), bmp(bmp_),
+			messageId(messageId_), message(nullptr) { }
+		Area(const Common::Rect &areaPos_, unsigned int messageId_, const char *record_ = nullptr) :
+			areaPos(areaPos_), record(record_), bmp(nullptr), messageId(messageId_), message(nullptr) { }
+	} areas[] = {
+		Area(Common::Point(174, 181), "APL.bmp", 74),
```

Comment:
Why do you want to create all these objects on heap instead of making them const and moving outside of the method?

Reply:
I wanted to avoid calling constructor at startup of scummvm. It's an array of objects with constructors that will be called at init time.